tests/pytests/test_vecsim.py

```diff
@@ -1291,3 +1291,35 @@ def test_redisearch_memory_limit():
     # reset env (for clean RLTest run with env reuse)
     env.assertTrue(conn.execute_command('CONFIG SET', 'maxmemory', '0'))
     env.expect('FT.CONFIG', 'SET', 'VSS_MAX_RESIZE', '0').ok()
+
+def test_rdb_memory_limit():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    env.skipOnCluster()
+    conn = getConnectionByEnv(env)
+
+    used_memory = int(env.cmd('info', 'memory')['used_memory'])
+    maxmemory = used_memory * 5
+    block_size = maxmemory // (16*4) // 2 # half of memory limit divided by blob size
+
+    # succeed to create indexes with no limits
+    env.expect('FT.CREATE', 'idx-flat', 'SCHEMA', 'v', 'VECTOR', 'FLAT', '10', 'TYPE', 'FLOAT32',
+               'DIM', '16', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 100, 'BLOCK_SIZE', block_size).ok()
+    env.expect('FT.CREATE', 'idx-hnsw', 'SCHEMA', 'v', 'VECTOR', 'HNSW', '8', 'TYPE', 'FLOAT32',
+               'DIM', '128', 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', 1000000).ok()
+    # sets memory limit
+    env.assertTrue(conn.execute_command('CONFIG SET', 'maxmemory', maxmemory))
+
+    # The actual test: try creating indexes from rdb.
+    # should succeed after changing initial cap and block size to 0 and default
+    env.assertTrue(conn.execute_command('DEBUG', 'RELOAD'))
+
+    info_data = to_dict(env.cmd("FT.DEBUG", "VECSIM_INFO", "idx-flat", "v"))
+    env.assertNotEqual(info_data['BLOCK_SIZE'], block_size)
+    # TODO: if we ever add INITIAL_CAP to debug data, add check here
+    # TODO: uncomment when BLOCK_SIZE is added to FT.CREATE on HNSW
+    # info_data = to_dict(env.cmd("FT.DEBUG", "VECSIM_INFO", "idx-hnsw", "v"))
+    # env.assertNotEqual(info_data['BLOCK_SIZE'], block_size)
```

Comment:
I think you can test it by modifying `VSS_MAX_RESIZE` before reloading

Reply:
It won’t do, because the upper limit of 1024 on the block size by default.